Library/OcDeviceMiscLib/SetResizableBar.c

```diff
@@ -309,8 +381,234 @@ SetResizableBarOnDevice (
+  UINT32                                                   Offset;
+  UINT32                                                   Index;
+  UINT32                                                   ResizableBarNumber;
+  UINT64                                                   Capabilities;
+  UINT64                                                   NewCapabilities;
+  UINT32                                                   OldBar[PCI_MAX_BAR];
+  UINT32                                                   NewBar[PCI_MAX_BAR];
+  INTN                                                     Bit;
+
+  Status = LocatePciCapabilityRbIo (
+             PciRootBridgeIo,
+             PciAddress,
+             PCI_EXPRESS_EXTENDED_CAPABILITY_RESIZABLE_BAR_ID,
+             &ResizableBarOffset
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "OCDM: RBAR is unsupported by device - %r\n", Status));
+    return EFI_UNSUPPORTED;
+  }
+
+  ResizableBarControl.Uint32 = 0;
+  Offset                     = ResizableBarOffset + sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_HEADER)
+                               + sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_CAPABILITY);
+  Status = PciRootBridgeIo->Pci.Read (
+                                  PciRootBridgeIo,
+                                  EfiPciWidthUint8,
+                                  PciAddrOffset (PciAddress, Offset),
+                                  sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_CONTROL),
+                                  &ResizableBarControl
+                                  );
+
+  DEBUG ((
+    DEBUG_INFO,
+    "OCDM: RBAR control is %X, total %u - %r\n",
+    ResizableBarControl.Uint32,
+    MIN (ResizableBarControl.Bits.ResizableBarNumber, PCI_MAX_BAR),
+    Status
+    ));
+
+  if (EFI_ERROR (Status)) {
+    return EFI_UNSUPPORTED;
+  }
+
+  ResizableBarNumber = MIN (ResizableBarControl.Bits.ResizableBarNumber, PCI_MAX_BAR);
+
+  Status = PciRootBridgeIo->Pci.Read (
+                                  PciRootBridgeIo,
+                                  EfiPciWidthUint8,
+                                  PciAddrOffset (PciAddress, ResizableBarOffset + sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_HEADER)),
+                                  sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_ENTRY) * ResizableBarNumber,
+                                  Entries
+                                  );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "OCDM: RBAR caps cannot be read - %r\n", Status));
+    return EFI_UNSUPPORTED;
+  }
+
+  Status = PciRootBridgeIo->Pci.Read (
+                                  PciRootBridgeIo,
+                                  EfiPciWidthUint32,
+                                  PciAddrOffset (PciAddress, OFFSET_OF (PCI_TYPE00, Device.Bar)),
+                                  PCI_MAX_BAR,
+                                  (VOID *)OldBar
```

Comment:
Sorry - another one - for consistency if we're removing `(VOID *)` in one place where it's not needed, I guess we should remove it in all calls to `Pci.Read` or `Pci.Write` in this file where we are passing a pointer to a `VOID *` parameter.

Reply:
(I'm picking up on this because @xCuri0 said that he got the code which he was asked to change from the original code!)

Reply:
You can if you want to, but that's really a separate thing. About all the "it's in the original code" comments - just because things slip review once, doesn't mean they should slip twice. With regards to casts, my personal policy to have them if and only if there is a noteworthy effect. E.g., an integer value may be shortened, signed-to-unsigned conversion, etc. All pointers cast implicitly to VOID *, there's no concerns with alignment or strict aliasing, it's as boring as it gets.

Reply:
I agree if things slip once they shouldn't slip twice, maybe my suggestion was misunderstood, my suggestion is to fix them in the other places where they also apply, at least in this file, now that they have been spotted.

Reply:
@mikebeaton I could fix the PciIo code but I'm unable to test it due to my system having a broken PciIo implementation. For that reason I will probably just leave PciIo as is

Reply:
Okay. I'll close all these and often a separate PR applying the same minor fixes already applied in your new bits. Thank you!